lib/CMakeLists.txt

```diff
@@ -26,37 +27,18 @@ set(OPTSCHED_SRCS Scheduler/aco.cpp
   Wrapper/OptSchedMachineWrapper.cpp
   Wrapper/OptSchedDDGWrapperBasic.cpp
   Wrapper/OptSchedGenericTarget.cpp
-)
+  Wrapper/AMDGPU/GCNOptSched.cpp
+  Wrapper/AMDGPU/OptSchedGCNTarget.cpp
+  Wrapper/AMDGPU/OptSchedDDGWrapperGCN.cpp
 
-set(OPTSCHED_TARGET_DEPS LLVMCodeGen)
+  STATIC
+)
 
-if(OPTSCHED_ENABLE_AMDGPU)
-  list(APPEND OPTSCHED_SRCS
-    Wrapper/AMDGPU/GCNOptSched.cpp
-    Wrapper/AMDGPU/OptSchedGCNTarget.cpp
-    Wrapper/AMDGPU/OptSchedDDGWrapperGCN.cpp
-  )
-  if(TARGET AMDGPUCommonTableGen)
-    list(APPEND OPTSCHED_TARGET_DEPS AMDGPUCommonTableGen)
-  endif()
-  if(llvm_subproject)
-    include_directories(
-      ${LLVM_MAIN_SRC_DIR}/lib/Target/AMDGPU
-      ${LLVM_BINARY_DIR}/lib/Target/AMDGPU
-    )
-  endif()
-endif()
+#set(OPTSCHED_EXTRA_DEFINITIONS "-DINSERT_ON_STEPFRWRD")
 
-function(add_optsched_library name)
-  if(LLVM_VERSION VERSION_LESS 8.0)
-    add_llvm_loadable_module(${name} OBJECT ${ARGN})
-  else()
-    add_llvm_library(${name} MODULE OBJECT ${ARGN})
-  endif()
-endfunction()
+#add_dependencies(OptSched ${OPTSCHED_TARGET_DEPS})
 
-add_optsched_library(OptSched
-  ${OPTSCHED_SRCS}
-  LINK_LIBS ${OPTSCHED_LINK_LIBS}
-)
-add_dependencies(OptSched ${OPTSCHED_TARGET_DEPS})
+target_compile_definitions(LLVMOptSched PRIVATE INSERT_ON_STEPFRWRD)
+target_include_directories(LLVMOptSched PRIVATE ${OPTSCHED_TARGET_DEPS})
+target_compile_options(LLVMOptSched PRIVATE -Wno-suggest-override)
```

Comment:
Don't include things like this in `target_compile_options`. Pass that in on the commandline or use one of CMake's ways to make it more configurable (e.g. toolchain files, presets, ...).